network/p2p/middleware.go

```diff
@@ -340,6 +351,17 @@ func (m *Middleware) stop() {
 //
 // Dispatch should be used whenever guaranteed delivery to a specific target is required. Otherwise, Publish is
 // a more efficient candidate.
+//
+// The following benign errors can be returned from libp2p:
+// - the peer ID for the target node ID cannot be found.
+// - the msg size exceeds result returned from unicastMaxMsgSize(msg)
+// - the libP2P node fails to publish the message.
+// - the libP2P node fails to create the stream.
+// - setting write deadline on the stream fails.
+// - the gogo protobuf writer fails to write the message.
+// - flushing the stream fails.
```

Comment:
I think this list is overly specific and leaks too much information about the implementation. You could simplify it to something like this without losing too much context.

```suggestion
// The following benign errors can be returned:
// - he peer ID for the target node ID cannot be found.
// - the msg size was too large.
// - failed to send message to peer.
```

Reply:
https://github.com/onflow/flow-go/pull/2726/commits/3dac90936ad08f9dd47396507383ab559641c904